doc/cookbook/namedRoutes/NamedRoutesOperator.lhs

```diff
@@ -0,0 +1,259 @@
+
+Let's jump into the "movies" subtree node:
+
+
+``` haskell
+
+type ApiMovies =  "list" :> QueryParam "SortBy" SortBy :> Get '[JSON] [Movie]
+               :<|>  Capture "movieId" MovieId :> Get '[JSON] (Maybe Movie)
+               :<|>   Capture "movieId" MovieId :> ReqBody '[JSON] Movie :> Put '[JSON] NoContent
+               :<|> Capture "movieId" MovieId :> Delete '[JSON] Movie
+
+data SortBy = Year | Title
+
+instance ToHttpApiData SortBy where
+  toQueryParam Year = "year"
+  toQueryParam Title = "title"
+
+instance FromHttpApiData SortBy where
+  parseQueryParam "year" = Right Year
+  parseQueryParam "title" = Right Title
+  parseQueryParam param = Left $ param <> " is not a valid value"
+
+```
+So, remember, this type represents the `MoviesAPI` node that we’ve connected earlier to the main `API` tree.
+
+In this subtree, we illustrated both an endpoint with a **query param** and also, a **capture** with a subtree underneath it.
+
+So, let's go deeper into our API tree.
+
+
+As you can see, we end up implementing the deepest routes of our API.
+
+Small detail: as our main API tree is also a record, we need the `NamedRoutes` helper.
+To improve readability, we suggest you create a type alias:
+
+``` haskell
+type MovieCatalogAPI = API
+```
+
+That's it, we have our `MovieCatalogAPI` type!
+
+Let's make a server and a client out of it!
+
+### Server
+
+As you know, we can’t talk about a server, without addressing the handlers.
+
+First, we take our handlers…
+
+```haskell
+versionHandler :: Handler Version
+versionHandler = pure "0.0.1"
+
+movieListHandler :: Maybe SortBy -> Handler [Movie]
+movieListHandler _ = pure moviesDB
+
+moviesDB :: [Movie]
+moviesDB =
+  [ Movie "1" "Se7en" 1995
+  , Movie "2" "Minority Report" 2002
+  , Movie "3" "The Godfather" 1972
+  ]
+
+getMovieHandler :: MovieId -> Handler (Maybe Movie)
+getMovieHandler requestMovieId = go moviesDB
+  where
+    go [] = pure Nothing
+    go (movie:ms) | movieId movie == requestMovieId = pure $ Just movie
+    go (m:ms) = go ms
+
+updateMovieHandler :: MovieId -> Movie -> Handler NoContent
+updateMovieHandler requestedMovieId newMovie =
+   -- update the movie list in the database...
+   pure NoContent
+
+deleteMovieHandler :: MovieId -> Handler NoContent
+deleteMovieHandler _ =
+   -- delete the movie from the database...
+   pure NoContent
+
+```
+
+And assemble them together with the record structure, which is the glue here.
+
+```haskell
+server :: ServerT MovieCatalogAPI Handler
+server = versionHandler
+      :<|> movieListHandler
+      :<|> getMovieHandler
+      :<|> updateMovieHandler
+      :<|> deleteMovieHandler
+
+```
+As you might have noticed, we assemble our handlers with the same API’s records: `MoviesAPI` and `MovieAPI`.
+
+Remember the `mode` type parameter we saw earlier? Since we need to transform our API type into a _serve_r, we need to provide a server `mode`, which is `AsServerT` _handler_ here.
```

Comment:
```suggestion
Remember the `mode` type parameter we saw earlier? Since we need to transform our API type into a _server_, we need to provide a server `mode`, which is `AsServerT Handler` here.
```